### DISCUSSION TITLE
lab_3 tb error

### DISCUSSION THREAD

**Discussion by BarryJu1999** [bol-edu/HLS-SOC-Discussions#83] [Q&A] [answered]
如標題在使用makefile時，會出現error_coef的問題(下圖)

![error](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/86f2776d-5e62-46ab-94f3-aa73c04a4182)

將error_coef上移後，將bram include進設計中會出現syntax error的問題(下圖)

![bram](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/c9cb3d36-2ad9-43d2-b31c-98faff03db72)

將bram移出設計又會有以下問題(下圖)

![Do](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/eae3d4bc-a6b0-41b9-bf2e-9f0b6899c5d9)

想請問我應該怎麼做比較好QQ

**Comment by ruei7916** [comment]
看起來是bram11.v有錯
把第28行改成
`always @* Do = {32{EN}} & RAM[r_A>>2];    // read`
試試看

**Reply by BarryJu1999** [reply]
同學你好，非常謝謝你的幫助，更改後確實有用
但tb跑出新error(下圖)

![螢幕擷取畫面 2023-10-03 232604](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/0a2fb7f5-e024-4dff-8f67-0f3b5a19b1d1)

對應的程式碼部分為

![螢幕擷取畫面 2023-10-03 232437](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/47e91297-d3e7-46bc-bede-dcbf4b028ac3)

我不太確定這邊應該是否改成
ss_tdata和sm_tdata = ..... ;
尤其是sm_tdata是wire好像不能直接修改，
所以想另外請教這邊應該如何處理，謝謝同學。

**Reply by ruei7916** [reply]
檢查一下你fir_tb.v檔案中最後的task ss跟task sm是不是還在，他看起來像是沒找到這兩個task

**Reply by boledulab** [reply]
Thank ruei7916.  It should use {32{EN}} & RAM[r_A>>2}.  the Github is updated, but still use continuous assignment.

Jiin

**Reply by ruei7916** [reply, marked as answer]
@boledulab 
如果要用continuous assignment的話，第17行的`reg`要改成`wire`
順帶一提，`bram12.v`第28行也有相同的問題

**Comment by boledulab** [comment]
Yes, Do shoud be change to wire. Github updated. Thanks!
